app/controllers/activities_controller.rb

```diff
@@ -30,4 +30,38 @@ def poll
       format.js
     end
   end
+
+  private
+
+  def period_start
```

Comment:
Thoughts on adding an ActivitiesFilter service class? This would save the controller from having any specific DateTime logic

Reply:
Updated the controller and we don't need anymore #period_start and #period_end. I believe that since we only use #filter_activities() for filtering we can still leave that there to reduce overhead. If you still think it will be neater with and ActivitiesFilter class we can still go for it.